src/Internal/Service/GatewayService.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Class GatewayService
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Service;
+
+use WC_Order;
+use WC_Payment_Gateway_WCPay;
+use WCPay\Container;
+
+/**
+ * Service for accessing gateway methods.
+ *
+ * __This service is a workaround!__
+ * Various gateway methods, coming from core classes, cannot be
+ * extracted out of the gateway. Use this service to access them.
+ *
+ * The gateway here is a dynamic dependency, accessed through the
+ * container, instead of a constructor parameter. That's because
+ * the gateway depends on various services, and there would be a
+ * circular dependency otherwise.
+ *
+ * Before adding the method you need here, please consider
+ * whether it could be extracted out of the gateway instead.
+ */
+class GatewayService {
```

Comment:
For now this service is not needed, and I'd consider removing it. However, I don't want to delete it yet.